mesonbuild/build.py

```diff
@@ -2364,10 +2364,14 @@ def process_kwargs(self, kwargs):
                 # number of the version by default.
                 self.soversion = self.ltversion.split('.')[0]
             # macOS, iOS and tvOS dylib compatibility_version and current_version
-            self.darwin_versions = T.cast('T.Optional[T.Tuple[str, str]]', kwargs.get('darwin_versions'))
-            if self.darwin_versions is None and self.soversion is not None:
+            d = T.cast('T.Union[None, Literal[False], T.Tuple[str, str]]', kwargs.get('darwin_versions'))
```

Comment:
Shouldn't this be Optional[Literal[False]]?

Reply:
No, you either get `None` (the option was omitted), `False` (the option was explicitly an empty list), or `(str, str)`, the option was filled in and was not an empty list.

Reply:
And, `Optional[T]` is just a short hand for `Union[None, T]`, which is nice for a single value, but I think `Optional[Union[T, U]]` is less readable than `Union[None, T, U]`

Reply:
Right, fair enough, I misread this and thought it was a Union of 2 things in which case the shorthand is preferable.

Reply:
Although, maybe this would be made clearer with an Enum... Let me see how that looks.

Reply:
Eh, not as nice as I'd hoped, not really an better than what I have already